### PR TITLE
Adds Text Direction functionality to the TextField

### PR DESCRIPTION
**Pull request by jslavitz**
Adds a test to #17304 originally created by @xclud.

### CODE REVIEW

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -106,6 +106,7 @@ class TextField extends StatefulWidget {
     this.textCapitalization = TextCapitalization.none,
     this.style,
     this.textAlign = TextAlign.start,
+    this.textDirection = TextDirection.ltr,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/text_field.dart]: this should not have a default (or if it does, it should default to rtl)

File: packages/flutter/lib/src/material/text_form_field.dart [hunk 2]
```diff
@@ -58,6 +58,7 @@ class TextFormField extends FormField<String> {
     TextCapitalization textCapitalization = TextCapitalization.none,
     TextInputAction textInputAction,
     TextStyle style,
+    TextDirection textDirection = TextDirection.ltr,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/text_form_field.dart]: ditto

**Comment by Hixie** [comment]
cc @goderbauer for review

File: packages/flutter/lib/src/material/text_field.dart (hunk 1 above)
Comment (jslavitz) [reply packages/flutter/lib/src/material/text_field.dart]: Wait, right to left? Why would that be?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/text_field.dart (hunk 1 above)
Comment (tvolkert) [reply packages/flutter/lib/src/material/text_field.dart]: +1 for no default

Comment (xclud) [reply packages/flutter/lib/src/material/text_field.dart]: The time i did this PR i didn't know enumerations can have null value (I am a C# guy) and wished i could let the default value empty. This is the behavior i expected and i believe is correct.

@Hixie RTL default doesn't make any sense.

I am happy to see my PR is merged :)

Comment (Hixie) [reply packages/flutter/lib/src/material/text_field.dart]: RTL makes as much sense as LTR, but it forces people to realize why it's a bad idea to have a default. See https://master-docs-flutter-io.firebaseapp.com/flutter/dart-ui/TextDirection-class.html for a more detailed discussion.

**Pull request merged**